dashboard/app/models/unit_group_unit.rb

```diff
@@ -26,9 +26,15 @@ class UnitGroupUnit < ApplicationRecord
   # the experiment_name enabled.
   belongs_to :default_script, class_name: 'Unit', optional: true
 
+  after_destroy_commit :update_course_json
+
   def self.experiments
     Rails.cache.fetch("course_script_experiments") do
       UnitGroupUnit.where.not(experiment_name: nil).map(&:experiment_name)
     end
   end
+
+  def update_course_json
+    UnitGroup.find_by(id: course_id)&.write_serialization
```

Comment:
can you please test what happens when you delete a unit group which has units? It looks like you're covering this case with the `&.`, but it would be good to confirm.